### PR TITLE
[14.0][IMP] payroll: send email with payslip, fix payslip form view, better payslip line views

### PR DESCRIPTION
**Pull request by ghost**
This PR has 3 commits:
[IMP] payroll: send email with payslip
[FIX] payroll: payslip form view: name attributes for inheritance
[IMP] payroll: payslip line: better views, new field payslip_run_id

### CODE REVIEW

**Comment by nimarosa** [comment]
@appstogrow 

This looks great to me, if you run pre-commit before pushing it will pass the checks. Congrats on the email functionality, i was thinking in doing something like this time ago. 

I think runboat fails because you are missing the action (action_report_payslip) declaration.

**Comment by ghost** [comment]
Any idea why the positive/negative leave test doesn't create a Worked Days line with the leave?

**Comment by nimarosa** [comment]
> Any idea why the positive/negative leave test doesn't create a Worked Days line with the leave?

Maybe @mtelahun can answer this since she submitted the setting PR?

**Comment by mtelahun** [comment]
Arghh! Default resource.calendar defines a 5-day work week so the test fails if you run it on Saturday or Sunday.

**Comment by nimarosa** [comment]
@mtelahun maybe you can make a PR and make the test run with fixed dates? Or maybe you can create a full week calendar? 

Because that test will affect all PR tests if we make PRs in the weekend.

**Comment by nimarosa** [comment]
@appstogrow tests should be fixed with last merge. I think you gonna have to rebase.

**Comment by ghost** [comment]
It should be ready now.

File: payroll/views/hr_payslip_line_views.xml [hunk 1]
```diff
@@ -20,6 +20,7 @@
                 <field name="total" />
                 <field name="amount_select" invisible="1" />
                 <field name="register_id" invisible="1" />
+                <field name="payslip_run_id" />
```
Comment (nimarosa) [inline payroll/views/hr_payslip_line_views.xml]: I think we should keep this field invisible or better with the parameter optional="hide", so the user can show it or hide it depending of the situation. Payslip lines are often viewed from payslip form.

File: payroll/views/hr_payslip_line_views.xml [hunk 2]
```diff
@@ -105,11 +119,8 @@
     >
         <field name="name">Payslip Computation Details</field>
         <field name="res_model">hr.payslip.line</field>
-        <field name="view_mode">tree,form</field>
-        <field name="domain">[]</field>
-        <field name="context">
-            {'default_slip_id': active_id,'search_default_slip_id': active_id}
-        </field>
+        <field name="view_mode">tree,pivot,form</field>
+        <field name="domain">[('slip_id', 'in', active_ids)]</field>
```
Comment (nimarosa) [inline payroll/views/hr_payslip_line_views.xml:123]: Not sure what this change do?

Comment (nimarosa) [reply payroll/views/hr_payslip_line_views.xml:123]: the domain one

File: payroll/views/hr_payslip_line_views.xml [hunk 3]
```diff
@@ -65,6 +78,7 @@
                 />
                 <field name="amount_select" />
                 <field name="slip_id" />
+                <field name="payslip_run_id" />
```
Comment (nimarosa) [inline payroll/views/hr_payslip_line_views.xml:81]: Is there a case when the payslip_line_ids form is not displayed from the smart button in the payslip? There is no menu to access the full payslip_lines views. If i'm not wrong, then having a search for payslip_run_id will always return the same records in that tree view.

**Review by nimarosa — CHANGES_REQUESTED** [review]

File: payroll/views/hr_payslip_line_views.xml (hunk 2 above)
Comment (ghost) [reply payroll/views/hr_payslip_line_views.xml:123]: Select multiple payslips and click Action - Payslip Computation Details to see the lines of all the selected payslips (not just the first payslip). active_ids works in domain, but unfortunately not in context.

UPDATE: I don't need to change this action bound to Payslip. I created a new action bound to Payslip Batch.

File: payroll/views/hr_payslip_line_views.xml (hunk 3 above)
Comment (ghost) [reply payroll/views/hr_payslip_line_views.xml:81]: The user may use the Action menu to navigate to payslip lines list view. Then the user can search.

I just want to see pivot totals for a batch. So I made an action menu for Payslip Batch.

Comment (nimarosa) [reply payroll/views/hr_payslip_line_views.xml:81]: Great I didn't know that this action existed haha. Great

**Review by nimarosa — APPROVED** [review]

**Comment by nimarosa** [comment]
Okay so we are ready to merge this. 

/ocabot merge minor

**Pull request merged**